File: 2026/06-June/07-Sunday/solution.php
class Solution {

    function createBinaryTree($descriptions) {

        $nodes = [];
        $children = [];

        foreach ($descriptions as $d) {

            [$p, $c, $left] = $d;

            if (!isset($nodes[$p]))
                $nodes[$p] = new TreeNode($p);

            if (!isset($nodes[$c]))
                $nodes[$c] = new TreeNode($c);

            if ($left)
                $nodes[$p]->left = $nodes[$c];
            else
                $nodes[$p]->right = $nodes[$c];

            $children[$c] = true;
        }

        foreach ($descriptions as $d) {
            if (!isset($children[$d[0]]))
                return $nodes[$d[0]];
        }

        return null;
    }
}

File: 2026/06-June/07-Sunday/solution.py
class Solution:
    def createBinaryTree(self, descriptions):
        nodes = {}
        children = set()

        for p, c, left in descriptions:
            if p not in nodes:
                nodes[p] = TreeNode(p)

            if c not in nodes:
                nodes[c] = TreeNode(c)

            if left:
                nodes[p].left = nodes[c]
            else:
                nodes[p].right = nodes[c]

            children.add(c)

        for p, _, _ in descriptions:
            if p not in children:
                return nodes[p]
